src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -750,4 +750,26 @@ public ModelAndView redirectGetSolutionRepositoryFiles(@PathVariable Long exerci
 
         return new ModelAndView("forward:/api/repository/" + participation.getId() + "/files-content");
     }
+
+    /**
+     * GET programming-exercise/:exerciseId/solution-file-names
+     *
+     * Returns the solution repository file names for a given programming exercise.
+     * Note: This endpoint redirects the request to the ProgrammingExerciseParticipationService. This is required if
+     * the solution participation id is not known for the client.
+     * @param exerciseId the exercise for which the solution repository files should be retrieved
+     * @return a redirect to the endpoint returning the files with content
+     */
+    @GetMapping(SOLUTION_REPOSITORY_FILE_NAMES)
+    @PreAuthorize("hasRole('TA')")
+    @FeatureToggle(Feature.ProgrammingExercises)
+    public ModelAndView redirectGetSolutionRepositoryFilesWithoutContent(@PathVariable Long exerciseId) {
```

Comment:
It looks like you did not add any Unit/Integration tests for this endpoint. Please consider adding one.

Reply:
Implemented